README.md

```diff
@@ -2,23 +2,163 @@
+
+
+## Compiling
+
+You should only need to compile if you want to modify to the IPLBoot source code. Otherwise, you can just download needed files from the [latest release](https://github.com/redolution/iplboot/releases/latest).
+
+1. Ensure you have the latest version of [devkitPro](https://devkitpro.org/wiki/Getting_Started) installed.
+2. Ensure you have the latest version of [libogc2](https://github.com/extremscorner/libogc2) installed (see [instructions](#installing-libogc2) below).
+3. Ensure your devkitPro environment variables are set: `DEVKITPRO` and `DEVKITPPC`
+
+IPLBoot also acts as a server for @emukidid's [usb-load](https://github.com/emukidid/gc-usb-load), should you want to use it for development purposes.
+
+### PicoBoot
+
+1. Run `make dol`.
+2. Follow usage instructions above using the newly created `build/iplboot.dol`.
+
+### Qoob Pro
+
+1. Obtain the gc-pal-10 IPL ROM (MD5: `0cdda509e2da83c85bfe423dd87346cc`) and copy it to the root of this project as `ipl.rom`. This is currently the only IPL that has been verified to work.
+2. Run `make qoobpro`.
+3. Follow usage instructions above using the newly created `build/iplboot.gcb`.
+
+### Qoob SX
+
+NOTE: Qoob SX is currently nonfunctional as the resulting BIOS is too large.
+
+1. Ensure dolxz is installed (see [instructions](#installing-dolxz) below).
+2. Ensure doltool is installed (see [instructions](#installing-doltool) below).
+3. Obtain `qoob_sx_13c_upgrade.elf` and copy it to the root of this project. (TODO: MD5 and more details about the file.)
+4. Run `make qoobsx`.
+5. Follow usage instructions above using the newly created `build/qoob_sx_iplboot_upgrade.elf`
+
+### Viper
+
+1. Run `make viper`.
+2. Follow usage instructions above using the newly created `build/iplboot.vgc`.
+
+### GameCube Memory Card
+
+1. Ensure dol2gci is installed (see [instructions](#installing-dol2gci) below).
+2. Run `make gci`.
+3. Follow usage instructions above using the newly created `build/iplboot.gci`.
+
+*Optional:* If you want to reduce the file size by ~50% so it will take up less space on your memory card, you can create a compressed version.
+
+1. Ensure dol2gci is installed (see [instructions](#installing-dol2gci) below).
+2. Ensure dolxz is installed (see [instructions](#installing-dolxz) below).
+3. Run `make gci_compressed`.
+4. Follow usage instructions above using the newly created `build/iplboot_xz.gci` (Notice the `_xz` suffix).
+
+### Compressed DOL
+
+Should you need it, you can create a compressed version of the base DOL.
+
+1. Ensure dolxz is installed (see [instructions](#installing-dolxz) below).
+2. Run `make dol_compressed`.
+3. Follow usage instructions above using the newly created `build/iplboot_xz.dol` (Notice the `_xz` suffix).
+
+
+## Dependencies
+
+**NOTE:** Some of these are required only in certain scenarios and you may not need them. Check the instructions for your target device first.
+
+### Installing libogc2
```

Comment:
I'm a bit iffy on providing installation instructions for our dependencies.
I don't see Python 3 listed anywhere, but it's required for all the modchips. I know it's likely to already be installed on most Linux machines, but definitely not on Windows.

Reply:
You are right. I think we should just link to the repos/download. I'll dump this section and add python to the list.

Should we just add dolxz, dol2gci, and doltool to the repo? Not sure that is best practice, but they are small and it would be convenient.

Reply:
I don't mind vendoring build tool binaries, that would certainly make things easier. Just make sure to link the source code.

Reply:
#12 adds binaries to repo. Will remove instructions for installing these.